### DISCUSSION TITLE
Credit for contributors

### DISCUSSION THREAD

**Discussion by JimiHFord** [tteck/Proxmox#1222] [General]
**Describe the issue**
It looks like I wasn't credited as a contributor to the project even though my Omada PR https://github.com/tteck/Proxmox/pull/96 was merged a while back. Could this be because of the way the maintainer had handled the PR? I'm not trying to throw shade 😄, just trying to figure out what caused this.

**Screenshots**
Visiting https://github.com/tteck/Proxmox/graphs/contributors you'll see that @JimiHFord doesn't show up 😢 

**Resolution**
Could something be done to make sure I show up as a contributor? I am a HUGE fan of this project, and I want to keep contributing provided that I receive some small token of recognition.

**Comment by JimiHFord** [comment]
lol so I just fixed bug #1221 and now I'm showing up as a contributor! But I'm wondering if the author of the Omada commits shows up as @tteck 🤔  - haven't looked yet

**Reply by tteck** [reply]
Maybe the PR never made it to the main branch.

**Reply by tteck** [reply]
Or, maybe to be a Contributor, it has to go directly to the main branch?

**Comment by tteck** [comment]
It seems that if you make a commit to the pull-requests branch and I pull it into the main branch, you will be listed as a co-author. https://github.com/tteck/Proxmox/commit/f8c38b8177002b059d36c2dd826d514627b92232 I've noticed that Sergio is not listed in the contributors' list. Unfortunately, I don't know how to rectify this without pulling directly into the main branch, which I am hesitant to do.

**Reply by JimiHFord** [reply]
Unfortunately, co-author doesn't give full recognition to contributors. Right now, @Donkeykong307 is the only one that shows up as having made a significant contribution because his PR was merged direclty into `main`.

<img width="454" alt="Screenshot 2023-03-29 at 4 17 24 PM" src="https://user-images.githubusercontent.com/1879448/228658149-783eb479-7188-4f0b-a8bf-e5d7c20c8dd4.png">

The only thing that shows up for me is my one-liner fix that got merged directly to `main`. My Omada contribution #96 doesn't show up.
<img width="448" alt="Screenshot 2023-03-29 at 4 21 58 PM" src="https://user-images.githubusercontent.com/1879448/228658415-ffd277b1-73c2-4f4f-a33b-fc1e31075921.png">

https://github.com/tteck/Proxmox/graphs/contributors

Could you please consider altering your workflow to something like this:

* Contributor opens a PR (let's say PR#69, for instance) from `contrib-username-fork:pr-branch` ➡️ `origin:main`
* Maintainer(s) tests the PR 
  * your workflow could look something like
    * `git add remote contrib-username-fork git@github.com:contrib-username-fork/Proxmox.git`
    * `git fetch contrib-username-fork/pr-branch`
    * `git checkout contrib-username-fork/pr-branch`
    * `./dev-utils/swap-urls.sh <all-the> <arguments-required> <to-update-the> <scripts-for-testing-on> <pr-69>`
    * `git push -u origin pr-branch:pr-69`
    * run manual tests that test PR#69 in isolation of any other proposed changes
* If tests pass, maintainer(s) merge PR#69 into `origin:main` and delete `origin:pr-69`

I am happy to whip together a [sed](https://www.howtogeek.com/666395/how-to-use-the-sed-command-on-linux/) script and open a PR that does what you'd expect `./dev-utils/swap-urls.sh` to do.

Not only will this give full recognition to contributors, but it will easily allow contributors to test their changes using their fork since they could use `./dev-utils/swap-urls.sh` to test on their own fork and branch. Also, it would allow a maintainer to test multiple PRs in parallel in complete isolation of each other (e.g. pr-69, pr-70).

As an added bonus, if you adopt a workflow that attributes full recognition to contributors, I'll open up a PR that checks future pull requests for bash syntax errors via GitHub Actions. 😍

**Comment by tteck** [comment]
I take it that you missed last weeks commit https://github.com/tteck/Proxmox/commit/d3b7ccc33324d57b98034b17e80f87fecebdb6a4 to the `pull_request_template.md`

**Reply by JimiHFord** [reply]
I saw that, but I didn't interpret that as a license to start opening up PRs against `main`. Thank you for your accommodation 🥳 !!
